packages/xod-patch-search/test/index.spec.js

```diff
@@ -0,0 +1,89 @@
+import path from 'path';
+import { assert } from 'chai';
+import { listPatches } from 'xod-project';
+import { loadProject } from 'xod-fs';
+
+import { createIndex, createIndexData } from '../src/index';
+
+describe('xod-patch-search/index', () => {
+  let idx = {};
+  const workspace = path.resolve(__dirname, '../../../workspace');
+  const getProjectPath = projectName => path.resolve(workspace, projectName);
+
+  before(async () => {
+    const project = await loadProject(getProjectPath('welcome-to-xod'));
+    const indexData = createIndexData(listPatches(project));
+    idx = createIndex(indexData);
+  });
+
+  it('searches by path correctly', () => {
+    const result = idx.search('path:number');
+    assert.equal(
+      result[0].item.path,
+      'xod/core/format-number'
+    );
+  });
+
+  it('searches by lib correctly', () => {
+    const result = idx.search('lib:xod/core');
+    assert.lengthOf(result, 63);
```

Comment:
Very fragile assertion. xod/core constantly grows. I think we should count patch count programmatically. The readability would degrade, but the test will be solid.